### PR TITLE
Implement Decorator Pattern for sensor, cleanser, and prediction -> including: addition of classes, modification of necessary tests.

### LINKED ISSUE #185: Decorator

### PR DESCRIPTION
**Pull request by Birdbh**
Fixes #185

### CODE REVIEW

File: src/sensors/DecoratorPattern.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class SensorDataProcessor:
```
Comment (JamesShadowfyre) [inline src/sensors/DecoratorPattern.py:1]: Can you add some specification to this decorator pattern class? Update with comments explaining requires/modifies/effects

File: src/sensors/DecoratorPattern.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    def process_data(self):
+        pass
+
+class SensorDataDecorator(SensorDataProcessor):
+    def __init__(self, sensor):
+        self.sensor = sensor
+
+    def __getattr__(self, name):
+        "Delegate attribute access to the decorated sensor object"
```
Comment (JamesShadowfyre) [inline src/sensors/DecoratorPattern.py:10]:
```suggestion
       # "Delegate attribute access to the decorated sensor object"
```

Comment (JamesShadowfyre) [reply src/sensors/DecoratorPattern.py:10]: Make clear its a comment

**Review by JamesShadowfyre — CHANGES_REQUESTED** [review]
This design pattern needs some documentation. Please update with some requires and modifies comments so it is understood how to use this pattern for future additions to the code.

**Comment by Birdbh** [comment]
@JamesShadowfyre Based on your feedback I have added Requirments, Modifies, and Effects to fully explain how the decorator pattern is implemented and its intended functionality

**Review by brkykrclk — APPROVED** [review]
Thank god sensor.py didnt change this time, thank you for the update

**Review by Reebs296 — APPROVED** [review]
Looks good to me, check checkkkkk!

**Review by Birdbh — COMMENTED** [review]
Changes are fixed, modifies/requires/effects put together

**Review by Jcp149 — APPROVED** [review]
tests pass, looks great, i think the design pattern will do this project very well in the long run and make it easier to comprehend the object!

**Review by brkykrclk — APPROVED** [review]
Ok cool sensor.py still same thank youu

**Pull request merged**